diesel/src/pg/expression/functions.rs

```diff
@@ -62,3 +63,46 @@ define_sql_function! {
     #[cfg(feature = "postgres_backend")]
     fn set_masklen<T: InetOrCidr + SingleValue>(addr: T, len: Integer) -> T;
 }
+
+define_sql_function! {
+    /// Returns the lower bound of the range.
+    /// if the range is empty or has no lower bound, it returns NULL.
+    /// # Example
+    ///
+    /// ```rust
+    /// # include!("../../doctest_setup.rs");
+    /// #
+    /// # table! {
+    /// #     posts {
+    /// #         id -> Integer,
+    /// #         versions -> Range<Integer>,
+    /// #     }
+    /// # }
+    /// #
+    /// # fn main() {
+    /// #     run_test().unwrap();
+    /// # }
+    /// #
+    /// # fn run_test() -> QueryResult<()> {
+    /// #     use self::posts::dsl::*;
+    /// #     use std::collections::Bound;
+    /// #     let conn = &mut establish_connection();
+    /// #     diesel::sql_query("DROP TABLE IF EXISTS posts").execute(conn).unwrap();
+    /// #     diesel::sql_query("CREATE TABLE posts (id SERIAL PRIMARY KEY, versions INT4RANGE NOT NULL)").execute(conn).unwrap();
+    /// #
+    /// use diesel::dsl::lower;
+    /// diesel::insert_into(posts)
+    ///     .values(&[
+    ///        versions.eq((Bound::Included(5), Bound::Included(7))),
+    ///        versions.eq((Bound::Unbounded, Bound::Included(7)))
+    ///     ]).execute(conn)?;
+    ///
+    /// let cool_posts = posts.select(lower(versions))
+    ///     .load::<Option<i32>>(conn)?;
+    /// assert_eq!(vec![Some(5), None], cool_posts);
+    /// #     Ok(())
+    /// # }
+    /// ```
+    #[cfg(feature = "postgres_backend")]
+    fn lower<T: RangeHelper<Inner: SingleValue> + SingleValue>(range: T) -> Nullable<<T as RangeHelper>::Inner>;
```

Comment:
From what I could understand, `define_sql_function` does not accept `where` clause, so I used rust [1.79 associated item bounds syntax](https://blog.rust-lang.org/2024/06/13/Rust-1.79.0.html#bounds-in-associated-type-position) to make it inline, but this breaks the MNV

Reply:
I've pushed https://github.com/diesel-rs/diesel/pull/4103/commits/42681bd8d8b36f36c014f2c310ae158ac6cbb0bb to fix this.

Reply:
Thanks!